test/e2e/nodemaintenance_test.go

```diff
@@ -354,3 +362,74 @@ func checkFailureStatus(t *testing.T, f *framework.Framework) {
 		}
 	}
 }
+
+func getOperatorHostname(t *testing.T, f *framework.Framework) string {
+	pods := &corev1.PodList{}
+
+	err := f.Client.List(goctx.TODO(), &client.ListOptions{
+		LabelSelector: labels.SelectorFromSet(operatorLabel),
+	}, pods)
+
+	if err != nil {
+		t.Fatalf("failed listing operator pods: %v", err)
+	}
+
+	if len(pods.Items) < 1 {
+		t.Fatal("there are no operator pods")
+	}
+
+	return pods.Items[0].Spec.NodeName
+}
+
+func countWorkerNodes(t *testing.T, f *framework.Framework) int {
+	nodes := corev1.NodeList{}
+	labelSelector := labels.NewSelector()
+
+	requirement, err := labels.NewRequirement(masterLabelKey, selection.DoesNotExist, []string{})
+	if err != nil {
+		t.Fatalf("failed creating selector requirement: %v", err)
+	}
+
+	labelSelector = labelSelector.Add(*requirement)
+	err = f.Client.List(goctx.TODO(), &client.ListOptions{
+		LabelSelector: labelSelector,
+	}, &nodes)
+
+	if err != nil {
+		t.Fatalf("failed listing worker nodes: %v", err)
+	}
+
+	return len(nodes.Items)
+}
+
+func testDeploymentExistsOnNode(t *testing.T, f *framework.Framework, nodeName string) bool {
+	return sliceContainsString(getTestDeploymentHostnames(t, f), nodeName)
+}
+
+func getTestDeploymentHostnames(t *testing.T, f *framework.Framework) []string {
+	pods, err := getCurrentDeploymentPods(t, f)
+
+	if err != nil {
+		t.Fatalf("could not list test deployment pods: %v", err)
+	}
+
+	hostnames := []string{}
+
+	for _, pod := range pods.Items {
+		if !sliceContainsString(hostnames, pod.Spec.NodeName) {
+			hostnames = append(hostnames, pod.Spec.NodeName)
+		}
+	}
+
+	return hostnames
+}
+
+func sliceContainsString(slice []string, str string) bool {
```

Comment:
maybe create a utils folder/class for that under test folder and place this function there , as in:
https://github.com/kubevirt/node-maintenance-operator/blob/master/pkg/controller/nodemaintenance/utils.go

Reply:
Yea, sounds good, will do